src/NHibernate/Type/TypeFactory.cs

```diff
@@ -591,18 +592,12 @@ public static IType HeuristicType(string typeName, IDictionary<string, string> p
 			{
 				return null;
 			}
-			return HeuristicType(typeClass, parameters, length, false);
+
+			return HeuristicType(typeClass, parameters, length);
 		}
 
-		private static IType HeuristicType(System.Type typeClass, IDictionary<string, string> parameters, int? length, bool tryBasic = true)
+		private static IType HeuristicType(System.Type typeClass, IDictionary<string, string> parameters, int? length)
```

Comment:
Well. If you really don't like bool flags... :) But then I think this method needs to be renamed to something else.
To avoid confusion that some `HeuristicType` overrides lead to `GetBasicTypeByName` and some not.

Reply:
Agree.